packages/cdktf-cli/src/bin/cmds/helper/terraform-cloud-client.ts

```diff
@@ -39,6 +39,49 @@ export interface Workspaces {
   data: WorkspaceData[];
 }
 
+export interface ServiceDiscovery {
+  "modules.v1": string;
+  "providers.v1": string;
+  "motd.v1": string;
+  "state.v2": string;
+  "tfe.v2": string;
+  "tfe.v2.1": string;
+  "tfe.v2.2": string;
+  "versions.v1": string;
+}
+
+let cachedServiceDiscovery: ServiceDiscovery | undefined;
```

Comment:
I suspect it would be safer for the cache to take the hostname into account. I'm thinking about scenarios where users would have TFE and TFC configured and we make requests to both of them in the same process (e.g. when listing orgs that they have access to?)

Reply:
It's possible, but we'd have to refactor the init and terraform login to support scenarios like that. We only check for the supplied tfe hostname, or use app.terraform.io if not. When we start supporting multiple logins and checking the cached credentials to supply org names, we can revisit this.